src/OptionsResolver.php

```diff
@@ -0,0 +1,401 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver as BaseOptionsResolver;
+
+/**
+ * @internal
+ */
+final class OptionsResolver extends BaseOptionsResolver
```

Comment:
This (https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2500#discussion_r104912822) proposal would make a lot of code change in this PR (update of all configs definitions). Before doing that it would be great if one could double check the idea.

cc @SpacePossum , willing to take a look ?